code/extensions/Lumberjack.php

```diff
@@ -111,5 +112,21 @@ protected function shouldFilter() {
 		return $controller instanceof LeftAndMain
 			&& in_array($controller->getAction(), array("treeview", "listview", "getsubtree"));
 	}
+	
+	/**
+	 * Checks config for a specified child_classname on the class extending Lumberjack.
+	 * Uses SiteTree if none found in config.
+	 *
+	 * @return string
+	 */
+	protected function getChildClassName() {
+		$childClassName = "SiteTree";
+		if ($childClassNameConfig = Injector::inst()->create($this->owner->ClassName)->config()->child_classname) {

```

Comment:
Where does child_class come from? Also, this should use Injector::inst()->get() (or singleton($className)).